bytecomp/translmod.ml

```diff
@@ -88,22 +88,27 @@ let rec apply_coercion loc strict restr arg =
   match restr with
     Tcoerce_none ->
       arg
-  | Tcoerce_structure(pos_cc_list, id_pos_list) ->
+  | Tcoerce_structure(pos_cc_list, id_pos_list, runtime_fields) ->
+      assert (List.length runtime_fields = List.length pos_cc_list);
```

Comment:
this assertion is good to have, can you write a function `same_length` to save some time

Reply:
Easy to do, but would it actually save time? Both lists are always traversed to the end (unless it fails), and the only difference would be keeping count.

Reply:
Indeed, the benefit is marginal